conbench/app/index.py

```diff
@@ -1,25 +1,102 @@
-import flask as f
+import logging
+from collections import defaultdict
+from dataclasses import dataclass
+from urllib.parse import urlparse
+
+from sqlalchemy import select
 
 from ..app import rule
 from ..app._endpoint import AppEndpoint, authorize_or_terminate
 from ..app.benchmarks import RunMixin
 from ..config import Config
+from ..db import Session
+from ..entities.run import Run
+
+log = logging.getLogger(__name__)
 
 
 class Index(AppEndpoint, RunMixin):
-    def page(self, runs):
+    def page(self, repo_runs_map):
         return self.render_template(
             "index.html",
             application=Config.APPLICATION_NAME,
             title="Home",
-            runs=runs,
-            search_value=f.request.args.get("search"),
+            repo_runs_map=repo_runs_map,
+            # Note(JP): search_value is not consumed in template
+            # search_value=f.request.args.get("search"),
         )
 
     @authorize_or_terminate
     def get(self):
-        runs = self.get_display_runs()
-        return self.page(runs)
+        # Following
+        # https://docs.sqlalchemy.org/en/20/orm/queryguide/select.html#selecting-orm-entities
+        runs = Session.scalars(
+            select(Run).order_by(Run.timestamp.desc()).limit(1000)
+        ).all()
+
+        # Note(JP): group runs by associated commit.repository value Note that
+        # consistency between benchmark results in the run is currently not
+        # granted: https://github.com/conbench/conbench/issues/864
```

Comment:
I don't think we need to worry about this too much; I commented on that issue. Looks like you aren't worrying about it here beyond the comment. 🙂